### DISCUSSION TITLE
9월 19일 주간회의

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#1035] [General]
현재 진행중인 작업에 대해서 공유할 내용이 있으면 코멘트로 공유해주세요~

**Comment by al-bur** [comment]
## 내용
- 내 학습로그에서 뒤로가기를 3번해야 작동하는 문제 해결하여 PR 제출한 상태입니다.

https://github.com/woowacourse/prolog/pull/1034

**Comment by JUDONGHYEOK** [comment]
## 내용
- 프로필 뱃지 이미지 크게 렌더링 되는 문제 해결
- 기존 image-insert 브랜치 dev와 로컬에서 충돌 해결 및 ts 마이그레이션 중

**Comment by wonsss** [comment]
#1031
- 메인 페이지상 최신 레벨로그 조회 화면을 구현하여 PR 제출한 상태입니다. 
![image](https://user-images.githubusercontent.com/59413128/190957117-a385537f-9b4a-49dc-831c-c8f546c28b97.png)

**Comment by wishoon** [comment]
## 내용
* 레벨로그 댓글 화면 + 레벨로그 댓글 API 구현하여 PR 올라간 상태입니다.

#1032
#1033

**Reply by euijinkk** [reply]
프론트엔드 추가
- 비회원일 때에 댓글 보이도록 구현. (작성은 불가하게 버튼을 막아두었습니다.)

**Comment by her0807** [comment]
## 내용
- https://github.com/woowacourse/prolog/issues/1036

**Comment by Yboyu0u** [comment]
## 문서화 테스트 개선할 사항 분석

### 1. Document 와 NewDocument

- 현재 **RestAssured를 이용한 @SpringbootTest 통합 테스트인 Document**와 **MockMVC를 이용한 NewDocument**로 테스트가 나눠져 있다.
- **Document**를 사용하고 있는 테스트
    - 스터디로그, 프로필, 미션, 멤버, 로그인, 댓글, 태그 등등
    
   ![image](https://user-images.githubusercontent.com/63579113/190957598-fe99d5d7-9729-4942-937d-14cb649c1767.png)
    
- **NewDocument**를 사용하고 있는 테스트
    - 필터, 리포트, 세션, 스터디로그 역량 등등
    
  ![image](https://user-images.githubusercontent.com/63579113/190957635-26d5dc7f-dcd8-49df-9cf7-ac9f21d76a4c.png)
    

→ **사용 기준이 시간 순(개발이 디벨롭된 순서)이 아닌 것으로 보인다. (브리핑이 사전에 잘 이루어 지지 않았던 것으로 보인다.)**

Document를 사용 중인 문서화 테스트 중 comment와 sessiomMember은 내가 개발에 참여했기 때문. 

### **결론**

: **Document를 쓰던 지 NewDocument를 쓰던 지 하나만 쓰는 것이 올바르다고 판단된다.** 문서화 테스트라는 목적에 맞게 RestAssured에서 MockMvc로 개선을 시도했는데 이것이 잘 지켜지지 않고 있는 것으로 보이는데 NewDocument를 사용하도록 사전 브리핑이 필요할 것 같다. 

### 2. NewDocument에서 deplecated된 상수 사용

*`APPLICATION_JSON_UTF8_VALUE`*

![image](https://user-images.githubusercontent.com/63579113/190957724-fb955d5f-0619-4265-9e5b-8f793a93453f.png)

*`APPLICATION_JSON_VALUE` 로 수정하는 것이 옳아보인다. 이것으로 되어 있는 테스트도 있다. (한마디로 뒤죽박죽이다.)*

### 3.  가독성

- 예외 케이스에 대한 문서화는 이루어 지지 않고 있기 때문에 크게 문제가 되지는 않을 것으로 보이긴 한다.

**현재** 

```java
when(filterService.showAll(any()))
		.thenReturn(FILTER_RESPONSE_OF_NICKNAME_ASC_SORT);

given
    .when().get("/filters")
    .then().log().all().apply(document("filter/list")).statusCode(HttpStatus.OK.value());
```

**개선** 

```java
//given
when(filterService.showAll(any()))
     .thenReturn(FILTER_RESPONSE_OF_NICKNAME_ASC_SORT);
        
//when
ValidatableMockMvcResponse response = given
      .when().get("/filters")
      .then().log().all();

//then 
response.statusCode(HttpStatus.OK.value());
        
//docs
response.apply(document("filter/list"));
```

+**BDD mockito 적용(현재 mockito 사용중)**

```java
//given
given(filterService.showAll(any()))
      .willReturn(FILTER_RESPONSE_OF_NICKNAME_ASC_SORT);

//when
ValidatableMockMvcResponse response = given
      .when().get("/filters")
      .then().log().all();

//then 
response.statusCode(HttpStatus.OK.value());
        
//docs
response.apply(document("filter/list"));
```

**Reply by her0807** [reply]
오 ~ 영우~

**Comment by verus-j** [comment]
## 작업 내용
* docker compose를 이용해 로컬용 MySql 설정 완료
* #672 
## 이슈
* 애플리케이션 실행 시 초기 데이터 세팅을 위한 DataLoaderApplicationListener 실행 중 미션 저장 시 에러 발생

**Comment by EastHShin** [comment]
[DB 테이블 구조를 문서화한다. #1029](https://github.com/woowacourse/prolog/issues/1029)

DB 테이블이 추가되거나 변경 되는 상황에 대비해 문서화가 자동으로 되는 방법을 고안 해보겠습니다.
